client/pkg/testutil/before.go

```diff
@@ -38,3 +40,21 @@ func BeforeTest(t testing.TB) {
 
 	t.Cleanup(func() { assert.NoError(t, os.Chdir(path)) })
 }
+
+func BeforeIntegrationExamples(*testing.M) func() {
+	ExitInShortMode("Skipping: the tests require real cluster")
+
+	tempDir, err := ioutil.TempDir(os.TempDir(), "etcd-integration")
```

Comment:
Since 1.15 Go supports https://pkg.go.dev/testing#T.TempDir which provides unique directory each time it's called. Maybe we should use it instead writing our own tmp directory management?

Reply:
I do use it whenever possible. 
Put here we have `testing.M` (module), not `testing.T` (test) and the `testing.T` is not being given as parameters to 'Examples'.    So we are simulating test harness to be able to verify 'examples' does not not misbehave (due to 'stale' mocking) when running against real etcd.